### DISCUSSION TITLE
Using Supabase Auth outside of the Supabase API

### DISCUSSION THREAD

**Discussion by schaafenrico** [supabase/supabase#576] [Questions] [answered]
I would like to use the Supabase Auth to authorize custom stuff.
For example I have a next.js API route that returns a .csv file and I only want to allow certain people to download it. Since the supabase client stores the auth info in localstorage I guess I could just read it and send it with the request, but as far as I understand it could be that the access_token is invalid and I need to get a new one so I would need a new API route for that as well + the localstorage way does not work when I want to take advantage of getServerSideProps.

Are there and guidelines or recommendations how to handle these use cases?

**Comment by kiwicopple** [comment, marked as answer]
Hey @schaafenrico , did you see this example?

https://github.com/supabase/supabase/blob/master/examples/nextjs-with-supabase-auth/README.md

It shows how to user server-side auth with Next.js

**Reply by schaafenrico** [reply]
This totally answer's my question, dont know my I forgot to check examples. Thanks!
